engine/src/main/java/com/agiletec/aps/system/services/page/PageDAO.java

```diff
@@ -1,896 +1,894 @@
+            _logger.error("Error loading page records", t);
+            throw new RuntimeException("Error loading page records", t);
+        } finally {
+            closeDaoResources(res, stat);
+        }
+        return pages;
+    }
+
+    private void readPageWidgets(List<PageRecord> pages, boolean online, Connection conn) {
+        Statement stat = null;
+        ResultSet res = null;
+        try {
+            stat = conn.createStatement();
+            res = stat.executeQuery(online ? ALL_WIDGETS_ONLINE : ALL_WIDGETS_DRAFT);
+            PageRecord currentPage = null;
+            int currentWidgetNum = 0;
+            Widget[] currentWidgets = null;
+            Iterator<PageRecord> pagesIter = pages.iterator();
+            while (res.next()) {
+                String code = res.getString(1);
+                while (currentPage == null || !currentPage.getCode().equals(code)) {
+                    currentPage = pagesIter.next();
+                    currentWidgetNum = this.getWidgetArrayLength(online ? currentPage.getMetadataOnline() : currentPage.getMetadataDraft());
+                    currentWidgets = online ? currentPage.getWidgetsOnline() : currentPage.getWidgetsDraft();
+                }
+                this.readWidget(currentPage, currentWidgetNum, currentWidgets, 2, res);
+            }
+        } catch (Throwable t) {
+            _logger.error("Error loading page widgets - online/draft: {}", online, t);
+            throw new RuntimeException("Error loading page widgets - online/draft: " + online, t);
+        } finally {
+            closeDaoResources(res, stat);
+        }
+    }
+
+    protected int getWidgetArrayLength(PageMetadata metadata) {
+        int numFrames = -1;
+        if (metadata != null) {
+            PageModel model = metadata.getModel();
+            if (model != null) {
+                numFrames = model.getFrames().length;
+            }
+        }
+        return numFrames;
+    }
+
+    protected void readWidget(PageRecord page, int numOfFrames, Widget widgets[],
+            int startIndex, ResultSet res) throws ApsSystemException, SQLException {
+        Object posObj = res.getObject(startIndex);
+        if (posObj != null) {
+            int pos = res.getInt(startIndex);
+            if (pos >= 0 && pos < numOfFrames) {
+                Widget widget = this.createWidget(page, pos, res, startIndex + 1);
+                widgets[pos] = widget;
+            } else {
+                _logger.warn("The position read from the database exceeds the numer of frames defined in the model of the page {}", page
+                        .getCode());
+            }
+        }
+    }
+
+    protected PageRecord createPageRecord(String code, ResultSet res) throws Throwable {
+        PageRecord page = new PageRecord();
+        page.setCode(code);
+        page.setParentCode(res.getString(1));
+        page.setPosition(res.getInt(2));
+        if (res.getString(4) != null) {
```

Comment:
Double check this. I don't see a change in the number of columns selected in the query. The query changed but the column order looks the same?

Reply:
It's right. The fields changed are the following ones (from index 4)